pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -78,10 +78,20 @@ type Role string
 
 // K8SSelectorConfig is Kubernetes Selector Config
 type K8SSelectorConfig struct {
-	// +kubebuilder:validation:Required
-	Role  Role   `json:"role"`
-	Label string `json:"label,omitempty"`
-	Field string `json:"field,omitempty"`
+	// Role specifies the type of Kubernetes resource to limit the service discovery to.
+	// Accepted values are: Node, Pod, Endpoints, EndpointSlice, Service, Ingress.
+	// +required
+	Role Role `json:"role"`
```

Comment:
It would be good to have consts for the supported values:

```
const (
    KubernetesRolePod KubernetesRole = "Pod"
    KubernetesRoleService KubernetesRole = "Service"
    ...
```